Content.Shared/Item/ItemToggle/SharedItemToggleSystem.cs

```diff
@@ -0,0 +1,313 @@
+            // Starts the active sound (like humming).
+            if (TryComp(uid, out ItemToggleActiveSoundComponent? activeSound))
+            {
+                if (activeSound.ActiveSound != null && activeSound.PlayingStream == null)
+                {
+                    activeSound.PlayingStream = _audio.PlayPvs(activeSound.ActiveSound, uid, AudioParams.Default.WithLoop(true)).Value.Entity;
+                }
+            }
+
+            Activate(uid, itemToggle);
+            var ev = new ItemToggleActivatedEvent();
+            RaiseLocalEvent(uid, ref ev);
+        }
+        return true;
+    }
+
+    public bool TryDeactivate(EntityUid uid, EntityUid? user = null, ItemToggleComponent? itemToggle = null)
+    {
+        if (!Resolve(uid, ref itemToggle))
+            return false;
+
+        if (!itemToggle.Activated)
+            return true;
+
+        //Since there is currently no system that cancels a deactivation, it's all predicted.
+        var attempt = new ItemToggleDeactivateAttemptEvent(user);
+        RaiseLocalEvent(uid, ref attempt);
+
+        if (attempt.Cancelled && uid.Id == GetNetEntity(uid).Id)
+        {
+            return false;
+        }
+        else
+        {
+            _audio.PlayPredicted(itemToggle.SoundDeactivate, uid, user);
+
+            //Stops the active sound if there is any.
+            if (TryComp(uid, out ItemToggleActiveSoundComponent? activeSound))
+            {
+                activeSound.PlayingStream = _audio.Stop(activeSound.PlayingStream);
+            }
+
+            Deactivate(uid, itemToggle);
+
+            var ev = new ItemToggleDeactivatedEvent();
+            RaiseLocalEvent(uid, ref ev);
+
+            return true;
+        }
+    }
+
+    //Makes the actual changes to the item's components on activation.
+    private void Activate(EntityUid uid, ItemToggleComponent itemToggle)
+    {
+        itemToggle.Activated = true;
+
+        if (TryComp(uid, out ItemToggleSizeComponent? itemToggleSize))
+        {
+            UpdateItemComponent(uid, itemToggleSize);
+        }
+        if (TryComp(uid, out ItemToggleMeleeWeaponComponent? itemToggleMelee))
+        {
+            UpdateWeaponComponent(uid, itemToggleMelee);
+        }
+        UpdateAppearance(uid, itemToggle);
+        UpdateLight(uid, itemToggle);
+
+        Dirty(uid, itemToggle);
+    }
+    //Makes the actual changes to the item's components on deactivation.
+    private void Deactivate(EntityUid uid, ItemToggleComponent itemToggle)
+    {
+        itemToggle.Activated = false;
+
+        if (TryComp(uid, out ItemToggleSizeComponent? itemToggleSize))
+        {
+            UpdateItemComponent(uid, itemToggleSize);
+        }
+        if (TryComp(uid, out ItemToggleMeleeWeaponComponent? itemToggleMelee))
+        {
+            UpdateWeaponComponent(uid, itemToggleMelee);
+        }
+        UpdateAppearance(uid, itemToggle);
+        UpdateLight(uid, itemToggle);
+
+        Dirty(uid, itemToggle);
+    }
```

Comment:
Make the comments doc comments + also use event subs instead of trycomp just for cleanliness, same with the appearance and lights. You should be able to put the subs on the relevant systems i.e. meleeweaponsystem, itemsystem, wieldablesystem, etc.

Reply:
Done. I didn't add the subs to the appearance and light systems because those are in the robust toolbox. Kept those local. Same goes for playing the sounds.